packages/opentelemetry-plugin-grpc-js/src/utils.ts

```diff
@@ -44,3 +49,30 @@ export const grpcStatusCodeToCanonicalCode = (
 export const grpcStatusCodeToSpanStatus = (status: number): Status => {
   return { code: status };
 };
+
+/**
+ * Returns true if the metadata contains
+ * the opentelemetry outgoing request header.
+ */
+export const containsOtelMetadata = (metadata: grpcTypes.Metadata): boolean => {
+  return metadata.get(OTEL_OUTGOING_REQUEST_KEY).length > 0;
+};
+
+/**
+ * Returns true if the current plugin configuration
+ * ignores the given method.
+ */
+export const methodIsIgnored = (
+  methodName: string,
+  ignoredMethods?: string[]
+): boolean => {
+  if (!ignoredMethods) {
+    return false;
+  }
+  for (const pattern of ignoredMethods) {
+    if (new RegExp(`^${pattern}$`, 'i').test(methodName)) {
+      return true;
+    }
+  }
+  return false;
+};
```

Comment:
Could this be made like http plugin and support both partial regex and exact string matching?

https://github.com/open-telemetry/opentelemetry-js/blob/03981e4299282d0611d1255ab0076368d5830753/packages/opentelemetry-plugin-http/src/utils.ts#L140-L153

Reply:
Yep! I'll just use the exact same approach as in the http plugin